### ISSUE TITLE
FieldDropdown will change translation when drag the "sensing_of" block.

### ISSUE THREAD

**Issue by BrightLin** [LLK/scratch-blocks#2014]
### Expected Behavior

FieldDropdown can keep the text.

### Actual Behavior

The text of the dropdown field has changed.

### Steps to Reproduce

![backdrop](https://user-images.githubusercontent.com/15828392/69120300-41b5cf00-0ad4-11ea-88a8-346c50438939.gif)

![image](https://user-images.githubusercontent.com/15828392/69120356-64e07e80-0ad4-11ea-99c6-e5654ec1f9a0.png)

### Operating System and Browser

"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.70 Safari/537.36"

**Comment by apple502j** [comment]
Duplicate of LLK/scratch-gui#4209

**Comment by BrightLin** [comment]
@apple502j OK😀

**Comment by BrightLin** [comment]
@apple502j Hi, I found the cause of this issue:

### Can't get the source block when drag and create new sensor_of block. 
https://github.com/LLK/scratch-gui/blob/develop/src/lib/blocks.js
![image1](https://user-images.githubusercontent.com/15828392/72786038-98bc5980-3c67-11ea-8e10-1f83b0cef686.png)
`sensingOfBlock = vm.runtime.flyoutBlocks.getBlock(blockId);`

### Cannot get the block by id.
The result of "getBlock" may be undefined. The method 'getBlock' return 'this._blocks[id]'. But _blocks's keys can be id or opcode. 
![image2](https://user-images.githubusercontent.com/15828392/72786040-99ed8680-3c67-11ea-82ae-e1a9ddc89c53.png)

Modify the code like the following:

```js
ScratchBlocks.Blocks.sensing_of.init = function () {
        const blockId = this.id;

    //  fix #2014 Get the sensing_of block from vm.
    let defaultSensingOfBlock;
    const blocks = vm.runtime.flyoutBlocks._blocks;
    Object.keys(blocks).forEach((id) => {
      const block = blocks[id];
      if (id === blockType || block.opcode === blockType) {
        defaultSensingOfBlock = block;
      }
    });

        // Function that fills in menu for the first input in the sensing block.
        // Called every time it opens since it depends on the values in the other block input.
        const menuFn = function () {
            const stageOptions = [
                [ScratchBlocks.Msg.SENSING_OF_BACKDROPNUMBER, 'backdrop #'],
                [ScratchBlocks.Msg.SENSING_OF_BACKDROPNAME, 'backdrop name'],
                [ScratchBlocks.Msg.SENSING_OF_VOLUME, 'volume']
            ];
            const spriteOptions = [
                [ScratchBlocks.Msg.SENSING_OF_XPOSITION, 'x position'],
                [ScratchBlocks.Msg.SENSING_OF_YPOSITION, 'y position'],
                [ScratchBlocks.Msg.SENSING_OF_DIRECTION, 'direction'],
                [ScratchBlocks.Msg.SENSING_OF_COSTUMENUMBER, 'costume #'],
                [ScratchBlocks.Msg.SENSING_OF_COSTUMENAME, 'costume name'],
                [ScratchBlocks.Msg.SENSING_OF_SIZE, 'size'],
                [ScratchBlocks.Msg.SENSING_OF_VOLUME, 'volume']
            ];
            if (vm.editingTarget) {
                let lookupBlocks = vm.editingTarget.blocks;
                let sensingOfBlock = lookupBlocks.getBlock(blockId);

                // The block doesn't exist, but should be in the flyout. Look there.
                if (!sensingOfBlock) {
                    sensingOfBlock = vm.runtime.flyoutBlocks.getBlock(blockId) || defaultSensingOfBlock;
                    // If we still don't have a block, just return an empty list . This happens during
                    // scratch blocks construction.
                    if (!sensingOfBlock) {
                        return [['', '']];
                    }
                    // The block was in the flyout so look up future block info there.
                    lookupBlocks = vm.runtime.flyoutBlocks;
                }
```
